pkg/controller/stack/stack_controller.go

```diff
@@ -1220,27 +1227,74 @@ func (sess *reconcileStackSession) lookupPulumiAccessToken(ctx context.Context)
 	return "", false
 }
 
-// Make a working directory for building the given stack. These are stable paths, so that (for one
+// Make a root directory for the given stack, containing the home and workspace directories.
+func (sess *reconcileStackSession) MakeRootDir(ns, name string) (_path string, _err error) {
```

Comment:
If you don't plan to use the named return values, we generally don't have any.
```suggestion
func (sess *reconcileStackSession) MakeRootDir(ns, name string) (string, error) {
```

If you really want to use named return values, we don't need to include the `_` in names.

Reply:
I agree, that was due to copy-paste of the `MakeWorkspaceDir` func.